tests/unit/util.py

```diff
@@ -15,7 +15,7 @@ def skip_on_arch(min_arch=7):
         if torch.cuda.get_device_capability()[0] < min_arch:  #ignore-cuda
             pytest.skip(f"needs higher compute capability than {min_arch}")
     else:
-        assert deepspeed.accelerator.get_accelerator().device_name() == 'xpu'
+        assert deepspeed.accelerator.get_accelerator().device_name() in ['xpu', 'npu']
```

Comment:
Can we not use [DS_ACCELERATOR_LIST](https://github.com/microsoft/DeepSpeed/blob/1d1a20c5a1dd980438393500f14d8dc188f6258a/accelerator/real_accelerator.py#L53C10-L53C10) here?

Reply:
You are right, I will make [DS_ACCELERATOR_LIST](https://github.com/microsoft/DeepSpeed/blob/1d1a20c5a1dd980438393500f14d8dc188f6258a/accelerator/real_accelerator.py#L53C10-L53C10) as a public-accessible valid accelerator list, and we can check if current device is valid or not in all non-cuda logic branches.

Reply:
@tjruwase I expose `is_current_accelerator_supported()` in [commit](https://github.com/microsoft/DeepSpeed/pull/4569/commits/ee399db7655f6fc441edbcd9a91e8d9652970fc2) and I use it to assert.